103-proteccion-de-rutas

ByMofm opened the pull request:
-Se crea 2 componentes
-El componente ProtectedAdminRoute que se encarga de permitir el acceso si tiene rol = 1 (admin) y si no redirige al home

-El componente ProtectedUserRoute que se encarga de permitir el acceso a editar el perfil al usuario registrado unicamente, no permitiendo ingresar al backoffice.

Se reorganiza las rutas para que funcionen correctamente.

![pruebaproteccion](https://user-images.githubusercontent.com/85798648/169403989-8e4b5be1-2e2b-4c08-871b-5d363cdda19a.png)

Commit:
First Commit

-Se crea 2 componentes
-El componente ProtectedAdminRoute que se encarga de permitir el acceso si tiene rol = 1 (admin) y si no redirige al home

-El componente ProtectedUserRoute que se encarga de permitir el acceso a editar el perfil al usuario registrado unicamente, no permitiendo ingresar al backoffice.

ByMofm committed:
Merge branch 'main' into 103-Proteccion-de-rutas

Galaeno commented on src/components/Routes/ProtectedAdminRoutes.js:
```diff
@@ -0,0 +1,25 @@
+
+
+const useAuth = () => {
+    const _user=localStorage.getItem("roleId")
```
No se usa generalmente  guiones bajos para variables. Encima la variable dice user, pero el item dice roleid. Te recomiendo crear un archivo de constantes de roles el cual exportes un objeto con los roles ids para reutilizar y luego en éster archivo, hacer algo asi:
```js
const useAuth = () => {
    return {
        auth: isAdmin()
    };
}

const isAdmin = () => {
    const roleId = localStorage.getItem('roleId');

    return roleId === ROLES.ADMIN;
}
```

ByMofm replied:
Corregido

Galaeno commented on src/components/Routes/ProtectedUserRoute.js:
```diff
@@ -0,0 +1,25 @@
+
+
+const useAuth = () => {
+    const _user=localStorage.getItem("roleId")
```
Parecido al comenario sobre admin, pero la diferencia es que en vez de llamar a isAdmin, podría llamar a isLogged

ByMofm replied:
Corregido

Commit:
Merge branch 'main' into 103-Proteccion-de-rutas

Commit:
Update

-Se creo la constante /Roles
-Se modifico la protección tanto de admin como la de usuario para que compare esa constante con el roleId
